url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+				ResultData tmpResultData;
+				if(firstInBid.getVol() > firstInAsk.getVol()){
+					tmpResultData = new ResultData(firstInBid.getID(),firstInAsk.getID(),firstInAsk.getPrice(),firstInAsk.getVol(),tmpTime);
+					completedTrade.add(tmpResultData);
+					numberOfTrade++;
+					if(firstInBid.getID() < 0 || firstInAsk.getID() < 0){
+						strategyTrade.add(tmpResultData);
+					}
+					myBidList.updateFirst(firstInBid.getVol() - firstInAsk.getVol());
+					myAskList.deleteAtIndex(0);
+					numberOfTrade += matchTrade(myBidList,myAskList,completedTrade,tmpTime,strategyTrade);
+				}else if(firstInBid.getVol() == firstInAsk.getVol()){
+					tmpResultData = new ResultData(firstInBid.getID(),firstInAsk.getID(),firstInAsk.getPrice(),firstInBid.getVol(),tmpTime);
+					completedTrade.add(tmpResultData);
+					numberOfTrade++;
+					if(firstInBid.getID() < 0 || firstInAsk.getID() < 0){
+						strategyTrade.add(tmpResultData);
+					}
+					myAskList.deleteAtIndex(0);
+					myBidList.deleteAtIndex(0);
+				}else{
+					tmpResultData = new ResultData(firstInBid.getID(),firstInAsk.getID(),firstInAsk.getPrice(),firstInBid.getVol(),tmpTime);
+					completedTrade.add(tmpResultData);
+					numberOfTrade++;
+					if(firstInBid.getID() < 0 || firstInAsk.getID() < 0){
+						strategyTrade.add(tmpResultData);
+					}
+					myAskList.updateFirst(firstInAsk.getVol() - firstInBid.getVol());
+					myBidList.deleteAtIndex(0);
+				}
+			}
+		}
+		return numberOfTrade;
+	}
+
+	public void insertBidList(MyBidList myBidList, MyAskList myAskList,
+			LinkedList<ResultData> completedTrade, double tmpPrice, int tmpVol,
+			long tmpID, Time tmpTime) {
+		if(myAskList.getLength() > 0){
+			long tmpAskFirstID = myAskList.get(0).getID();
+			double tmpAskFirstPrice = myAskList.get(0).getPrice();
+			int tmpAskFirstVol = myAskList.get(0).getVol();
+			Time tmpAskFirstTime = myAskList.get(0).getTime();
+			double finishTime = tmpTime.getHours() + (tmpTime.getMinutes()/60.0);
+			if(tmpPrice >= tmpAskFirstPrice){
+				if(tmpVol > tmpAskFirstVol){
+					completedTrade.add(new ResultData(tmpID,tmpAskFirstID,tmpAskFirstPrice,tmpAskFirstVol,tmpTime));
+					myAskList.deleteAtIndex(0);
+					tmpAskFirstVol = tmpVol - tmpAskFirstVol;
+
+					//series3.add(finishTime,tmpPrice);
+					insertBidList(myBidList, myAskList,completedTrade, tmpPrice, tmpAskFirstVol ,tmpID, tmpTime);
+				}else if(tmpVol == tmpAskFirstVol){
+					completedTrade.add(new ResultData(tmpID,tmpAskFirstID,tmpAskFirstPrice,tmpVol,tmpTime));
+					myAskList.deleteAtIndex(0);
+					//series3.add(finishTime,tmpPrice);
+				}else{
+					completedTrade.add(new ResultData(tmpID,tmpAskFirstID,tmpAskFirstPrice,tmpVol,tmpTime));
+					tmpAskFirstVol -= tmpVol;
+					myAskList.updateFirst(tmpAskFirstVol);
+					//series3.add(finishTime,tmpPrice);
+				}
+			}else{
+				myBidList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+			}
+		}else{
+			myBidList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 671:700, 706:731.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.